File: android/src/main/kotlin/com/ottu/flutter/checkout/CheckoutArguments.kt
package com.ottu.flutter.checkout

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonIgnoreUnknownKeys

@Serializable
@JsonIgnoreUnknownKeys
data class CheckoutArguments(
    val merchantId: String,
    val apiKey: String,
    val sessionId: String,
    val amount: Double,
    val showPaymentDetails: Boolean,
    val apiTransactionDetails: String?,
    val formsOfPayment: List<String>?,
    val theme: CustomerTheme?,
    val paymentOptionsDisplaySettings: PaymentOptionsDisplaySettings,
    val payButtonText: PayButtonText?,
)

File: ios/Classes/CheckoutArguments.swift
//
//  CheckoutArguments.swift
//  ottu-flutter-checkout
//
//  Created by Vi on 10.10.2024.
//

struct CheckoutArguments: Decodable {
    let merchantId: String
    let apiKey: String
    let sessionId: String
    let amount: Double
    let showPaymentDetails: Bool
    let paymentOptionsDisplaySettings: PaymentOptionsDisplaySettings
    let apiTransactionDetails: String?
    let formsOfPayment: [String]?
    let theme: CustomerTheme?
}
